src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -303,12 +303,18 @@ internal void DisplayMachineLearningResults()
                     // DS Function node
                     if (result.Node.Type.NodeType.Equals(Function.FunctionNode))
                     {
+                        NodeSearchElement nodeSearchElement = null;
                         var element = zeroTouchSearchElements.FirstOrDefault(n => n.Descriptor.MangledName.Equals(result.Node.Type.Id));
 
-                        // Set PortToConnect for each element based on port-index and port-name
                         if (element != null)
                         {
-                            element.AutoCompletionNodeElementInfo = new AutoCompletionNodeElementInfo
+                            nodeSearchElement = (NodeSearchElement) element.Clone();
```

Comment:
So correct me if I am wrong here, we are now cloning the search element to have one instance for each port index that is returned in the ML results?

Reply:
We are cloning for each search suggestion as portToConnect is on each search element and it was being modified for same node suggestions.